src/diffusers/models/attention_processor.py

```diff
@@ -2087,43 +2087,55 @@ def __call__(self, attn: Attention, hidden_states: torch.FloatTensor, *args, **k
 
 class IPAdapterAttnProcessor(nn.Module):
     r"""
-    Attention processor for IP-Adapater.
+    Attention processor for Multiple IP-Adapater.
 
     Args:
         hidden_size (`int`):
             The hidden size of the attention layer.
         cross_attention_dim (`int`):
             The number of channels in the `encoder_hidden_states`.
-        num_tokens (`int`, defaults to 4):
+        num_tokens (List[`int`], defaults to [4]):
             The context length of the image features.
-        scale (`float`, defaults to 1.0):
+        scale (`float`, defaults to [1.0]):
             the weight scale of image prompt.
     """
 
-    def __init__(self, hidden_size, cross_attention_dim=None, num_tokens=4, scale=1.0):
+    def __init__(self, hidden_size, cross_attention_dim=None, num_tokens=(4,), scale=1.0):
         super().__init__()
 
         self.hidden_size = hidden_size
         self.cross_attention_dim = cross_attention_dim
+
+        if not isinstance(num_tokens, (tuple, list)):
+            raise ValueError("`num_tokens` should be a list of integers.")
         self.num_tokens = num_tokens
+
+        if not isinstance(scale, list):
+            scale = [scale] * len(num_tokens)
+        if len(scale) != len(num_tokens):
+            raise ValueError("`scale` should be a list of integers with the same length as `num_tokens`.")
         self.scale = scale
 
-        self.to_k_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
-        self.to_v_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
+        self.to_k_ip = nn.ModuleList(
+            [nn.Linear(cross_attention_dim, hidden_size, bias=False) for _ in range(len(num_tokens))]
+        )
+        self.to_v_ip = nn.ModuleList(
+            [nn.Linear(cross_attention_dim, hidden_size, bias=False) for _ in range(len(num_tokens))]
+        )
 
     def __call__(
         self,
         attn,
         hidden_states,
-        encoder_hidden_states=None,
+        encoder_hidden_states,
```

Comment:
I'd keep `encoder_hidden_states` optional here for backwards breaking

Reply:
ok 
but `encocer_hidden_state` was not actually an optional argument in the previous code, i.e. the code won't be able to run if `encoder_hidden_state` was passed as `None`, so I think it was a mistake in the signature. in that case it would not be a breaking change to remove `=None`, no?

Reply:
updated to follow closely with default attention processor